### DISCUSSION TITLE
[Docs]: Question about this tutorial: https://reactrouter.com/en/main/start/tutorial

### DISCUSSION THREAD

**Discussion by rebeccapeltz** [remix-run/react-router#11208] [Q&A] [answered]
### Describe what's incorrect/missing in the documentation

This is a great tutorial! I really learned a lot.  I tried to deploy it to GitHub pages and it always returns 200, but it returns the Error page.  I've been able to use gh-pages with other vite react apps.  When you deploy to gh-pages your path is prepended by the name of your repository and the code can't use that.  Is there a way to tell the router that there will be a prepended path when deployed?  I set the basepath in vite, but that doesn't help.  The code seems to work well locally if I take out the basepath.   I'm concerned about deployment.
I tried adding the basename to the router but it "oops". 
```
const router = createBrowserRouter([
  {
    basename: "react-router-tutorial",
    path: "/",
    element: <Root />,
    errorElement: <ErrorPage />,
```
<img width="1432" alt="image" src="https://github.com/remix-run/react-router/assets/354444/6ba6189c-60eb-460d-aedf-4e0ee51990b8">

**Comment by brophdawg11** [comment, marked as answer]
I think the issue is where you are configuring the `basename`.  It looks like it's part of your route definition, but it should be in the second option to `createBrowserRouter`.  Does the following work?

```js
const router = createBrowserRouter([
  {
    path: "/",
    element: <Root />,
    errorElement: <ErrorPage />,
  },
], {
    basename: "/react-router-tutorial",
});
```

**Reply by rebeccapeltz** [reply]
thanks!  
I had to put a `/` in front of the name for it to work - that matches the base in vite.config.js  `basename: "/react-router-tutorial" }`, but it's rendering on github pages now

**Reply by brophdawg11** [reply]
ah - right - `basename`'s should always start with the slash.  I updated my comment above to include the leading slash.  Glad it's working!
